### DISCUSSION TITLE
Is it possible to use react-snap with Gatsby?

### DISCUSSION THREAD

**Discussion by thejspro** [gatsbyjs/gatsby#29617] [Help]
I am not 100% happy with my page being rendered with JS. There are some critical pages on my website which should be fully rendered to HTML.

Currently the only way to achieve this is to hard-code the content inside a React component.

I have previously used [react-snap](https://github.com/stereobooster/react-snap) on another project (not Gatsby based) and loved it's simplicity and how it just works to produce clean, static HTML files with no JavaScript.

Is there a way to run react-snap post-build with Gatsby website?

Thank you!

**Comment by olivier-lacroix** [comment]
Hi @goodpixels , 

Gatsby already generates static HTML. You can see this by inspecting the source of the first URL you access on your site. After that first page is loaded, it is hydrated and turned into a SPA.

If you want to disable js entirely in the frontend, maybe have a look at [gatsby-plugin-no-javascript](https://www.gatsbyjs.com/plugins/gatsby-plugin-no-javascript/).
